Firestore/Swift/Source/Codable/EncoderDecoder.swift

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import FirebaseFirestore
+import FirebaseSharedSwift
+import Foundation
+
+public extension Firestore {
```

Comment:
Looks like the encoder can be done, but decoding may require the extra DocumentReference. Looking at workarounds

Reply:
Yeah, it could be a static property just fine.
The decoder technically technically doesn't need the overload that adds the document reference to the userInfo, it could be 'moved up' to the place where this is used. But it is just a tiny bit more of a breaking change since this API existed previously.

Let me know what you think. I think I prefer the static property too.

I think that we perhaps discussed the wrapper vs. static property a bit on the old PR discussion, but I may be mistaken.